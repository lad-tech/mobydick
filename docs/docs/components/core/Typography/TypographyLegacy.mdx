import Legacy from "./assets/Legacy.png";

<img src={Legacy} height={600} />

Reusable text

## Example

```tsx
import {TypographyLegacy} from "@lad-tech/mobydick-core";

export const Example = () => {
  return (
    <TypographyLegacy font={'Regular-Primary-S'}>Example</TypographyLegacy>
  );
};

```
## `Props`

### [`TextInput Props`](https://reactnative.dev/docs/textinput#props)

Inherits [TextInput Props](https://reactnative.dev/docs/textinput#props).

### `font`

| Type                                                                                                             | DEFAULT             |
|:-----------------------------------------------------------------------------------------------------------------|:--------------------|
| [TypographyLegacyProp](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/typography/types.ts#L49) | 'Regular-Primary-S' |

Font for text
