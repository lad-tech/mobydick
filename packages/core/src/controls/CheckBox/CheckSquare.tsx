import {FC} from 'react';

import {IPressableProps, Pressable} from '../../basic/components/Pressable';
import {ICommonControlProps} from '../types';
import {Check} from '../../styles/icons';
import useTheme from '../../styles/hooks/useTheme';
import px from '../../styles/utils/px';

const CheckSquare: FC<ICommonControlProps & IPressableProps> = ({
  selected,
  ...rest
}) => {
  const {colors} = useTheme();
  const width = rest.width || px(20);
  const height = rest.height || px(20);

  return (
    <Pressable {...rest}>
      {selected ? (
        <Check
          width={width}
          height={height}
          fill={rest.fill || colors.ElementAccent}
        />
      ) : null}
    </Pressable>
  );
};

export default CheckSquare;
