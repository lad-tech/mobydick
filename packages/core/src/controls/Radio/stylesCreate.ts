import {createStyles} from '../../styles';
import px from '../../styles/utils/px';

const stylesCreate = createStyles(
  ({spaces, colors}, selected: boolean, disabled: boolean) => ({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: 'transparent',
      opacity: disabled ? 0.4 : 1,
    },
    circle: {
      width: px(22),
      aspectRatio: 1,
      borderRadius: px(11),
      backgroundColor: selected ? colors.ElementAccent : 'transparent',
      borderColor: selected ? colors.ElementAccent : colors.BorderNormal,
      borderWidth: spaces.Space2,
      justifyContent: 'center',
      alignItems: 'center',
    },
    innerCircle: {
      width: px(18),
      aspectRatio: 1,
      borderWidth: spaces.Space2,
      borderRadius: px(9),
      borderColor: colors.BgPrimary,
    },
  }),
);

export default stylesCreate;
