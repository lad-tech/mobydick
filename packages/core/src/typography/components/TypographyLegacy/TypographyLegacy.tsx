import {FC} from 'react';

import Text from '../../../basic/components/Text/Text';
import {useFont} from '../../hooks';
import {IStyledTextLegacyProps} from '../../types';

export const TypographyLegacy: FC<IStyledTextLegacyProps> = ({
  children,
  font = 'Regular-Primary-S',
  style,
  ...props
}) => {
  const {fontStyle} = useFont(font);

  return (
    <Text style={[fontStyle, style]} {...props}>
      {children}
    </Text>
  );
};
