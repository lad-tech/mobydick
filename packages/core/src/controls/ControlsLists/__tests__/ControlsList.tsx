import {render} from '@testing-library/react-native';

import {ControlsList} from '../index';
import {CheckBox} from '../../CheckBox';
import {Radio} from '../../Radio';
import {TypographyLegacy} from '../../../typography/components/TypographyLegacy/TypographyLegacy';

describe('ControlsList', () => {
  it('should renders correctly with checkbox type', function () {
    const {toJSON} = render(
      <ControlsList single={false} onChange={jest.fn} values={[]}>
        <CheckBox value={'1'}>
          <TypographyLegacy font={'Regular-Primary-M'}>1</TypographyLegacy>
        </CheckBox>
        <CheckBox value={'2'}>
          <TypographyLegacy font={'Regular-Primary-M'}>2</TypographyLegacy>
        </CheckBox>
      </ControlsList>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly with checkbox type and single', function () {
    const {toJSON} = render(
      <ControlsList single onChange={jest.fn} values={[]}>
        <CheckBox value={'1'}>
          <TypographyLegacy font={'Regular-Primary-M'}>1</TypographyLegacy>
        </CheckBox>
        <CheckBox value={'2'}>
          <TypographyLegacy font={'Regular-Primary-M'}>2</TypographyLegacy>
        </CheckBox>
      </ControlsList>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly with checkbox type and horizontal', function () {
    const {toJSON} = render(
      <ControlsList single onChange={jest.fn} horizontal values={[]}>
        <CheckBox value={'1'}>
          <TypographyLegacy font={'Regular-Primary-M'}>1</TypographyLegacy>
        </CheckBox>
        <CheckBox value={'2'}>
          <TypographyLegacy font={'Regular-Primary-M'}>2</TypographyLegacy>
        </CheckBox>
      </ControlsList>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly with radio type', function () {
    const {toJSON} = render(
      <ControlsList single onChange={jest.fn} values={['1']}>
        <Radio value={'1'}>
          <TypographyLegacy font={'Regular-Primary-M'}>1</TypographyLegacy>
        </Radio>
        <Radio value={'2'}>
          <TypographyLegacy font={'Regular-Primary-M'}>2</TypographyLegacy>
        </Radio>
      </ControlsList>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly with radio type and single', function () {
    const {toJSON} = render(
      <ControlsList single onChange={jest.fn} values={[]}>
        <Radio value={'1'}>
          <TypographyLegacy font={'Regular-Primary-M'}>1</TypographyLegacy>
        </Radio>
        <Radio value={'2'}>
          <TypographyLegacy font={'Regular-Primary-M'}>2</TypographyLegacy>
        </Radio>
      </ControlsList>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should renders correctly with radio type and horizontal', function () {
    const {toJSON} = render(
      <ControlsList onChange={jest.fn} horizontal values={[]}>
        <Radio value={'1'}>
          <TypographyLegacy font={'Regular-Primary-M'}>1</TypographyLegacy>
        </Radio>
        <Radio value={'2'}>
          <TypographyLegacy font={'Regular-Primary-M'}>2</TypographyLegacy>
        </Radio>
      </ControlsList>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
