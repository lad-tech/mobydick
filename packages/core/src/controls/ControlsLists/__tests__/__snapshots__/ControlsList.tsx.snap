// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsList should renders correctly with checkbox type 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "maxWidth": "100%",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "rgba(21, 28, 44, 0.2)",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          undefined,
        ]
      }
      value="1"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "maxWidth": "100%",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "rgba(21, 28, 44, 0.2)",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          undefined,
        ]
      }
      value="2"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`ControlsList should renders correctly with checkbox type and horizontal 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "maxWidth": "100%",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "rgba(21, 28, 44, 0.2)",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          undefined,
        ]
      }
      value="1"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "maxWidth": "100%",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "rgba(21, 28, 44, 0.2)",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          undefined,
        ]
      }
      value="2"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`ControlsList should renders correctly with checkbox type and single 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "maxWidth": "100%",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "rgba(21, 28, 44, 0.2)",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          undefined,
        ]
      }
      value="1"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "maxWidth": "100%",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "borderColor": "rgba(21, 28, 44, 0.2)",
            "borderRadius": 4,
            "borderWidth": 2,
            "height": 20,
            "width": 20,
          },
          undefined,
        ]
      }
      value="2"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`ControlsList should renders correctly with radio type 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    needsOffscreenAlphaCompositing={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#2B78EE",
          "borderColor": "#2B78EE",
          "borderRadius": 11,
          "borderWidth": 2,
          "justifyContent": "center",
          "width": 22,
        }
      }
      value="1"
    >
      <View
        style={
          {
            "aspectRatio": 1,
            "borderColor": "#FFFFFF",
            "borderRadius": 9,
            "borderWidth": 2,
            "width": 18,
          }
        }
      />
    </View>
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    needsOffscreenAlphaCompositing={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "transparent",
          "borderColor": "rgba(21, 28, 44, 0.2)",
          "borderRadius": 11,
          "borderWidth": 2,
          "justifyContent": "center",
          "width": 22,
        }
      }
      value="2"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`ControlsList should renders correctly with radio type and horizontal 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    needsOffscreenAlphaCompositing={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "transparent",
          "borderColor": "rgba(21, 28, 44, 0.2)",
          "borderRadius": 11,
          "borderWidth": 2,
          "justifyContent": "center",
          "width": 22,
        }
      }
      value="1"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    needsOffscreenAlphaCompositing={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "transparent",
          "borderColor": "rgba(21, 28, 44, 0.2)",
          "borderRadius": 11,
          "borderWidth": 2,
          "justifyContent": "center",
          "width": 22,
        }
      }
      value="2"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;

exports[`ControlsList should renders correctly with radio type and single 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    needsOffscreenAlphaCompositing={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "transparent",
          "borderColor": "rgba(21, 28, 44, 0.2)",
          "borderRadius": 11,
          "borderWidth": 2,
          "justifyContent": "center",
          "width": 22,
        }
      }
      value="1"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    needsOffscreenAlphaCompositing={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "opacity": 1,
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "transparent",
          "borderColor": "rgba(21, 28, 44, 0.2)",
          "borderRadius": 11,
          "borderWidth": 2,
          "justifyContent": "center",
          "width": 22,
        }
      }
      value="2"
    />
    <Text
      allowFontScaling={true}
      style={
        [
          {
            "color": "#20242D",
            "fontFamily": "Inter-Regular",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 20,
            "minHeight": 20,
          },
          undefined,
        ]
      }
    >
      2
    </Text>
  </View>
</View>
`;
